openmdao/core/fileref.py

```diff
@@ -0,0 +1,94 @@
+"""
+Support for file variables.
+"""
+
+import sys
+import copy
+import os
+from six import iteritems
+
+#Public Symbols
+__all__ = ['FileRef']
+
+_file_meta = {
+    'binary': bool,
+}
+
+CHUNK = 1 << 20  # 1MB
+
+class FileRef(object):
+    """
+    A reference to a file on disk. As well as containing metadata information,
+    it supports :meth:`open` to read and write the file's contents.
+    """
+
+    def __init__(self, fname=None):
+        self.fname = fname
+        self.parent_dir = None
+        self.meta = {}
+
+    def __str__(self):
+        return "FileRef(%s): absolute: %s" % (self.fname, self._abspath())
+
+    def _set_meta(self, meta):
+        for name, typ in iteritems(_file_meta):
+            if name in meta:
+                self.meta[name] = typ(meta[name])
+
+    def open(self, mode):
+        """ Open file for reading or writing. """
+        if self.meta.get('binary') and 'b' not in mode:
+            mode += 'b'
+        return open(self._abspath(), mode)
+
+    def _abspath(self):
+        """ Return absolute path to file. """
+        if os.path.isabs(self.fname):
+            return self.fname
+        else:
+            return os.path.join(self.parent_dir, self.fname)
+
+    def validate(self, src_fref):
```

Comment:
what makes two files incompatible? different file names? Its not clear from the code. Comment this